dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -15,21 +15,28 @@ sealed class TypedColumn[T, U](
 )(
   implicit
   val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+) extends UntypedExpression[T] { self =>
 
   def this(column: Column)(implicit uencoder: TypedEncoder[U]) {
     this(FramelessInternals.expr(column))
   }
 
   def untyped: Column = new Column(expr)
 
-  def ===(other: U): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other)
-  }
+  def ===(other: U): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other)
+  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other.untyped)
 
-  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other.untyped)
-  }
+  def +(u: TypedColumn[T, U])(implicit n: Numeric[U]):  TypedColumn[T, U]= new TypedColumn[T, U](self.untyped.plus(u.untyped))
```

Comment:
Hey @adelbertc, you mean the type classes under `cats/kernel/instances/`? I love the idea of using something existing. However, if we define our own type classes, we can customize them to be compatible with the types supported by spark (say BigInt). One particularity of our case here, is that we don't really need the code inside these type classes, we just want to verify that what the user is doing will produce valid code in Spark's code generator.